src/main/java/bio/terra/service/snapshotbuilder/SelectAlias.java

```diff
@@ -1,22 +1,22 @@
 package bio.terra.service.snapshotbuilder;
 
-import bio.terra.service.snapshotbuilder.query.FieldVariable;
 import bio.terra.service.snapshotbuilder.query.SelectExpression;
+import bio.terra.service.snapshotbuilder.query.SqlExpression;
 import bio.terra.service.snapshotbuilder.query.SqlRenderContext;
 
 /** Wrap a field variable with an alias for use in a SELECT. */
 public class SelectAlias implements SelectExpression {
 
-  private final FieldVariable fieldVariable;
+  private final SqlExpression expression;
   private final String alias;
 
-  public SelectAlias(FieldVariable fieldVariable, String alias) {
+  public SelectAlias(SqlExpression expression, String alias) {
```

Comment:
Right now we have some more restricted types for different kinds of sql expressions, but no careful thought has been given to exactly what types are valid in what situations. If we wanted to be very specific, we'd have to look at the grammar for SQL and create types that match what the gramma requires. My thinking is to make things more permissive as needed until this work is done.